src/lib/input/input-container.ts

```diff
@@ -314,6 +354,30 @@ export class MdInputContainer implements AfterContentInit {
   /** Focuses the underlying input. */
   _focusInput() { this._mdInputChild.focus(); }
 
+  /** Whether the input container is in an error state. */
+  _isErrorState(): boolean {
+    const control = this._mdInputChild._ngControl;
+    const isInvalid = control ? control.invalid : false;
+    const isTouched = control ? control.touched : false;
+    const isSubmitted = this._parentForm ? this._parentForm.submitted : false;
```

Comment:
optional: rather than using a ternary, these could just be conditional expressions:

```ts
const isInvalid = control && control.invalid;
const isTouched = control && control.touched;
const isSubmitted = this._parentForm && this._parentForm.submitted;

return !!(isInvalid && (isTouched || isSubmitted));
```
(results in one boolean coercion instead of three)